### PR TITLE
Removed `Azure::Core::Http::InvalidHeaderException` and throw `std::invalid_argument` if the user provides invalid header arguments.

### PR DESCRIPTION
**Pull request by ahsonkhan**

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]

File: sdk/core/azure-core/src/http/http.cpp [hunk 1]
```diff
@@ -156,7 +156,7 @@ void Azure::Core::_detail::InsertHeaderWithValidation(
   {
```
Comment (JeffreyRichter) [inline sdk/core/azure-core/src/http/http.cpp:156]: This is a pretty substantial perf hit for every header for every operation invocation.
I didn't know we were doing this; I don't think I would have had it.

Comment (ahsonkhan) [reply sdk/core/azure-core/src/http/http.cpp:156]: I agree, but we need to let the caller know they have some bug in their code.

I wish there was a mechanism for us to mark APIs like `response.SetHeader()` internal-only. Right now this is a public API, that users can call and pass in any string. Validating that is useful. It's possible the users pass in some untrusted strings here.

Alternatively, we should consider trying to scope down the API surface and limit the several ways folks pass in data.

Let's discuss it separate from this PR.

Comment (vhvb1989) [reply sdk/core/azure-core/src/http/http.cpp:156]: I never liked this T.T , and still don't like it.

Headers most likely added by service clients. The service client should catch an invalid header during testing the sdk and update the header name to be valid.

Same thing could apply with end-users, and expect them to use the service response as enough to realize they set an invalid header.

An application running on production won't be able to do anything about this exception.

File: sdk/core/azure-core/src/http/raw_response.cpp [hunk 2]
```diff
@@ -26,7 +26,7 @@ void RawResponse::SetHeader(uint8_t const* const first, uint8_t const* const las
 
   if (end == last)
   {
-    throw InvalidHeaderException("Invalid header. No delimiter ':' found.");
+    throw std::invalid_argument("Invalid header. No delimiter ':' found.");
```
Comment (vhvb1989) [inline sdk/core/azure-core/src/http/raw_response.cpp:29]: In this case we want to throw `TrasnportException` and not invalid arg.  RawResponse comes only from a transport request.

Comment (ahsonkhan) [reply sdk/core/azure-core/src/http/raw_response.cpp:29]: `RawResponse` is a public type in a public header.
`SetHeader` is a public API on that type that accept some parameters.
This check is validating those inputs.

**Pull request merged**